docs/src/sequencepaxos/storage.md

```diff
@@ -0,0 +1,133 @@
+```rust,edition2018,no_run,noplaypen
+    // from the module omnipaxos_storage::memory_storage
+    #[derive(Clone)]
+    pub struct MemoryStorage<T, S>
+    where
+        T: Entry,
+        S: Snapshot<T>,
+    {
+        /// Vector which contains all the replicated entries in-memory.
+        log: Vec<T>,
+        /// Last promised round.
+        n_prom: Ballot,
+        /// Last accepted round.
+        acc_round: Ballot,
+        /// Length of the decided log.
+        ld: u64,
+        ...
+    }
+
+    impl<T, S> Storage<T, S> for MemoryStorage<T, S>
+    where
+        T: Entry,
+        S: Snapshot<T>,
+    {
+        fn append_entry(&mut self, entry: T) -> u64 {
+            self.log.push(entry);
+            self.get_log_len()
+        }
+
+        fn append_entries(&mut self, entries: Vec<T>) -> u64 {
+            let mut e = entries;
+            self.log.append(&mut e);
+            self.get_log_len()
+        }
+
+        fn append_on_prefix(&mut self, from_idx: u64, entries: Vec<T>) -> u64 {
+            self.log.truncate(from_idx as usize);
+            self.append_entries(entries)
+        }
+
+        fn set_promise(&mut self, n_prom: Ballot) {
+            self.n_prom = n_prom;
+        }
+
+        fn set_decided_idx(&mut self, ld: u64) {
+            self.ld = ld;
+        }
+
+        fn get_decided_idx(&self) -> u64 {
+            self.ld
+        }
+
+        fn set_accepted_round(&mut self, na: Ballot) {
+            self.acc_round = na;
+        }
+
+        fn get_accepted_round(&self) -> Ballot {
+            self.acc_round
+        }
+
+        fn get_entries(&self, from: u64, to: u64) -> Vec<T> {
+            self.log
+                .get(from as usize..to as usize)
+                .unwrap_or(&[])
+                .to_vec()
+        }
+
+        fn get_log_len(&self) -> u64 {
+            self.log.len() as u64
+        }
+
+        fn get_suffix(&self, from: u64) -> Vec<T> {
+            match self.log.get(from as usize..) {
+                Some(s) => s.to_vec(),
+                None => vec![],
+            }
+        }
+
+        fn get_promise(&self) -> Ballot {
+            self.n_prom
+        }
+        ...
+    }
+```
+
+## PersistentStorage
+`PersistentStorage` is a persistent storage implementation that stores the replicated log and the state of OmniPaxos. The struct uses [Commitlog](https://crates.io/crates/commitlog) to store replicated logs, for storing the OmniPaxos state users can choose between [rocksDB](https://crates.io/crates/rocksdb) or [sled](https://crates.io/crates/sled). Users can configure the path to log entries and OmniPaxos state, and storage-related options through `PersistentStorageConfig`. The configuraiton struct features a `default()` constructor for generating default configuration, and the constructor `with()` that takes path and storage options as arguments. We recommend the user to generate the default configuration and then setting the desired fields.
```

Comment:
```suggestion
`PersistentStorage` is a persistent storage implementation that stores the replicated log and the state of OmniPaxos. The struct uses [Commitlog](https://crates.io/crates/commitlog) to store the replicated log, and the state is stored on [sled](https://crates.io/crates/sled) by default. The state can be changed to be stored on [RocksDB](https://crates.io/crates/rocksdb) instead of sled by using the feature`rocksdb`. Users can configure the path to log entries and OmniPaxos state, and storage-related options through `PersistentStorageConfig`. The configuration struct features a `default()` constructor for generating default configuration, and the constructor `with()` that takes path and storage options as arguments. 
```